tests_new/integration_tests/modules/s3_datasets/test_s3_tables_columns.py

```diff
@@ -7,35 +8,81 @@
     list_dataset_table_columns,
 )
 from integration_tests.errors import GqlError
+from integration_tests.modules.s3_datasets.conftest import (
+    TABLES_FIXTURES_PARAMS,
+    TABLES_CONFIDENTIALITY_FIXTURES_PARAMS,
+)
 
 log = logging.getLogger(__name__)
 
 
-def test_sync_dataset_table_columns():
-    # TODO
-    pass
+@pytest.mark.parametrize(*TABLES_FIXTURES_PARAMS)
+def test_sync_dataset_table_columns(client1, tables_fixture_name, request):
+    tables = request.getfixturevalue(tables_fixture_name)
+    response = sync_dataset_table_columns(client1, tables[0].tableUri)
+    assert_that(response.count).is_equal_to(3)
+    assert_that(response.nodes[0].name).is_equal_to('column1')
+    assert_that(response.nodes[0].typeName).is_equal_to('int')
 
 
-def test_sync_dataset_table_columns_unauthorized():
-    # TODO
-    pass
+@pytest.mark.parametrize(
+    'dataset_fixture_name,tables_fixture_name',
+    [('session_s3_dataset1', 'session_s3_dataset1_tables')],
+)
+def test_sync_dataset_table_columns_unauthorized(client2, dataset_fixture_name, tables_fixture_name, request):
+    dataset = request.getfixturevalue(dataset_fixture_name)
+    tables = request.getfixturevalue(tables_fixture_name)
+    table_uri = tables[0].tableUri
+    assert_that(sync_dataset_table_columns).raises(GqlError).when_called_with(client2, table_uri).contains(
```

Comment:
Does it actually catcher GqlError? When I test locally (with AWS deployment) the same thing returns just Exception(

Reply:
Did you test it in your deployment (not locally)?

Reply:
Yes, and the tests for the table_columns pass:

![image](https://github.com/user-attachments/assets/4e78086f-761b-417f-b088-a807f217f21b)


There is an issue for the update_persisten_env which I am looking into. It should not be related to this, but I am waiting to merge until I get the issue
![image](https://github.com/user-attachments/assets/5a0f9b03-e35f-4b68-a894-50023cf6d926)

Reply:
The error on the teardown of resources might be related to the metadata forms delete:
<img width="1265" alt="image" src="https://github.com/user-attachments/assets/578d9194-07d6-4817-99af-57b51016e9c5">
The error in the test of the persistent environment is a Timeout
![Uploading image.png…]()

Reply:
Worksheet is not yet connected to MFs. In any way. That's strange